[CI] ShardBulkInferenceActionFilterIT testRestart {p0=false p1=true} failing

**Build Scans:**
- [elasticsearch-periodic-platform-support #10241 / amazonlinux-2023_platform-support-aws](https://gradle-enterprise.elastic.co/s/fglvst5752ehg)
- [elasticsearch-periodic-platform-support #10192 / oraclelinux-8_platform-support-unix](https://gradle-enterprise.elastic.co/s/gj5chrresofpm)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:inference:internalClusterTest" --tests "org.elasticsearch.xpack.inference.action.filter.ShardBulkInferenceActionFilterIT.testRestart {p0=false p1=true}" -Dtests.seed=9641C2BAED105BC7 -Dtests.locale=sd -Dtests.timezone=Asia/Taipei -Druntime.java=24
```

**Applicable branches:**
9.1

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.inference.action.filter.ShardBulkInferenceActionFilterIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testRestart%20%7Bp0%3Dfalse%20p1%3Dtrue%7D'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: Failed to index document MGfUuDdINcJupKXdXHRZ: org.elasticsearch.index.mapper.DocumentParsingException: [1:772] failed to parse field [_inference_fields] of type [_inference_fields] in document with id 'MGfUuDdINcJupKXdXHRZ'. Preview of field's value: 'null'
```

**Issue Reasons:**
- [9.1] 2 failures in test testRestart {p0=false p1=true} (1.1% fail rate in 179 executions)
- [9.1] 2 failures in pipeline elasticsearch-periodic-platform-support (25.0% fail rate in 8 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

Full stack trace: 
```
REPRODUCE WITH: ./gradlew ":x-pack:plugin:inference:internalClusterTest" --tests "org.elasticsearch.xpack.inference.action.filter.ShardBulkInferenceActionFilterIT.testRestart {p0=false p1=true}" -Dtests.seed=9641C2BAED105BC7 -Dtests.locale=sd -Dtests.timezone=Asia/Taipei -Druntime.java=24
  2> java.lang.AssertionError: Failed to index document MGfUuDdINcJupKXdXHRZ: org.elasticsearch.index.mapper.DocumentParsingException: [1:772] failed to parse field [_inference_fields] of type [_inference_fields] in document with id 'MGfUuDdINcJupKXdXHRZ'. Preview of field's value: 'null'
        at __randomizedtesting.SeedInfo.seed([9641C2BAED105BC7:3CC2EDB68F8EFD2]:0)
        at org.elasticsearch.test.ESTestCase.fail(ESTestCase.java:2714)
        at org.elasticsearch.xpack.inference.action.filter.ShardBulkInferenceActionFilterIT.assertRandomBulkOperations(ShardBulkInferenceActionFilterIT.java:276)
        at org.elasticsearch.xpack.inference.action.filter.ShardBulkInferenceActionFilterIT.testRestart(ShardBulkInferenceActionFilterIT.java:227)

        Caused by:
        org.elasticsearch.index.mapper.DocumentParsingException: [1:772] failed to parse field [_inference_fields] of type [_inference_fields] in document with id 'MGfUuDdINcJupKXdXHRZ'. Preview of field's value: 'null'
            at org.elasticsearch.index.mapper.FieldMapper.rethrowAsDocumentParsingException(FieldMapper.java:241)
            at org.elasticsearch.index.mapper.FieldMapper.parse(FieldMapper.java:194)
            at org.elasticsearch.index.mapper.DocumentParser.parseObjectOrField(DocumentParser.java:471)
            at org.elasticsearch.index.mapper.DocumentParser.doParseObject(DocumentParser.java:539)
            at org.elasticsearch.index.mapper.DocumentParser.parseObject(DocumentParser.java:527)
            at org.elasticsearch.index.mapper.DocumentParser.innerParseObject(DocumentParser.java:381)
            at org.elasticsearch.index.mapper.DocumentParser.parseObjectOrNested(DocumentParser.java:342)
            at org.elasticsearch.index.mapper.DocumentParser.internalParseDocument(DocumentParser.java:153)
            at org.elasticsearch.index.mapper.DocumentParser.parseDocument(DocumentParser.java:99)
            at org.elasticsearch.index.mapper.DocumentMapper.parse(DocumentMapper.java:128)
            at org.elasticsearch.index.shard.IndexShard.prepareIndex(IndexShard.java:1092)
            at org.elasticsearch.index.shard.IndexShard.applyIndexOperation(IndexShard.java:1019)
            at org.elasticsearch.index.shard.IndexShard.applyIndexOperationOnPrimary(IndexShard.java:963)
            at org.elasticsearch.action.bulk.TransportShardBulkAction.executeBulkItemRequest(TransportShardBulkAction.java:420)
            at org.elasticsearch.action.bulk.TransportShardBulkAction$2.doRun(TransportShardBulkAction.java:277)
            at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:27)
            at org.elasticsearch.action.bulk.TransportShardBulkAction.performOnPrimary(TransportShardBulkAction.java:345)
            at org.elasticsearch.action.bulk.TransportShardBulkAction.dispatchedShardOperationOnPrimary(TransportShardBulkAction.java:182)
            at org.elasticsearch.action.bulk.TransportShardBulkAction.dispatchedShardOperationOnPrimary(TransportShardBulkAction.java:83)
            at org.elasticsearch.action.support.replication.TransportWriteAction$1.doRun(TransportWriteAction.java:245)
            at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:27)
            at org.elasticsearch.common.util.concurrent.TimedRunnable.doRun(TimedRunnable.java:35)
            at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:1067)
            at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:27)
            at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1095)
            at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:619)
            at java.lang.Thread.run(Thread.java:1447)

            Caused by:
            java.lang.IllegalArgumentException: featureValue must be a positive normal float, got: -2.1474836E9 for feature feature_4 on field other_field.inference.chunks.embeddings which is less than the minimum positive normal float: 1.1754944E-38
                at org.apache.lucene.document.FeatureField.setFeatureValue(FeatureField.java:162)
                at org.apache.lucene.document.FeatureField.<init>(FeatureField.java:147)
                at org.elasticsearch.index.mapper.vectors.SparseVectorFieldMapper.parse(SparseVectorFieldMapper.java:392)
                at org.elasticsearch.xpack.inference.mapper.SemanticTextFieldMapper.parseCreateFieldFromContext(SemanticTextFieldMapper.java:664)
                at org.elasticsearch.xpack.inference.mapper.SemanticInferenceMetadataFieldsMapper.parseCreateField(SemanticInferenceMetadataFieldsMapper.java:154)
                at org.elasticsearch.index.mapper.FieldMapper.parse(FieldMapper.java:192)
                ... 25 more
```

--- a/x-pack/plugin/inference/qa/test-service-plugin/src/main/java/org/elasticsearch/xpack/inference/mock/TestSparseInferenceServiceExtension.java
+++ b/x-pack/plugin/inference/qa/test-service-plugin/src/main/java/org/elasticsearch/xpack/inference/mock/TestSparseInferenceServiceExtension.java
@@ -32,6 +32,7 @@ import org.elasticsearch.inference.UnifiedCompletionRequest;
 import org.elasticsearch.inference.WeightedToken;
 import org.elasticsearch.inference.configuration.SettingsConfigurationFieldType;
 import org.elasticsearch.rest.RestStatus;
+import org.elasticsearch.search.lookup.FieldLookup;
 import org.elasticsearch.xcontent.ToXContentObject;
 import org.elasticsearch.xcontent.XContentBuilder;
 import org.elasticsearch.xpack.core.inference.results.ChunkedInferenceEmbedding;
@@ -191,7 +192,9 @@ public class TestSparseInferenceServiceExtension implements InferenceServiceExte
 
         private static float generateEmbedding(String input, int position) {
             // Ensure non-negative and non-zero values for features
-            return Integer.toUnsignedLong(input.hashCode()) + 1 + position;
+            int hash = input.hashCode();
+            int absHash = (hash == Integer.MIN_VALUE) ? Integer.MAX_VALUE : Math.abs(hash);
+            return absHash + 1.0f + position;
         }
 
         public static class Configuration {

--- a/x-pack/plugin/inference/qa/test-service-plugin/src/main/java/org/elasticsearch/xpack/inference/mock/TestSparseInferenceServiceExtension.java
+++ b/x-pack/plugin/inference/qa/test-service-plugin/src/main/java/org/elasticsearch/xpack/inference/mock/TestSparseInferenceServiceExtension.java
@@ -32,7 +32,6 @@ import org.elasticsearch.inference.UnifiedCompletionRequest;
 import org.elasticsearch.inference.WeightedToken;
 import org.elasticsearch.inference.configuration.SettingsConfigurationFieldType;
 import org.elasticsearch.rest.RestStatus;
-import org.elasticsearch.search.lookup.FieldLookup;
 import org.elasticsearch.xcontent.ToXContentObject;
 import org.elasticsearch.xcontent.XContentBuilder;
 import org.elasticsearch.xpack.core.inference.results.ChunkedInferenceEmbedding;

--- a/x-pack/plugin/inference/qa/test-service-plugin/src/main/java/org/elasticsearch/xpack/inference/mock/TestSparseInferenceServiceExtension.java
+++ b/x-pack/plugin/inference/qa/test-service-plugin/src/main/java/org/elasticsearch/xpack/inference/mock/TestSparseInferenceServiceExtension.java
@@ -191,7 +191,7 @@ public class TestSparseInferenceServiceExtension implements InferenceServiceExte
 
         private static float generateEmbedding(String input, int position) {
             // Ensure non-negative and non-zero values for features
-            return Math.abs(input.hashCode()) + 1 + position;
+            return Integer.toUnsignedLong(input.hashCode()) + 1 + position;
         }
 
         public static class Configuration {

